spec/deaf_grandma_spec.rb

```diff
@@ -9,10 +9,12 @@
   end
   
   it "says 'NOT SINCE 1964!' when we yell" do
-    #implement your test here
+    expect(script.speak("HELLO GRANDMA!")).to eq "NOT SINCE 1964!"
   end
 
   it "EXTRA CREDIT: How would you test yelling BYE?" do
-    #implement your test here
+    grandma = DeafGrandma.new
+    grandma.speak("BYE")
+    expect(grandma.instance_variable_get(:@bye_counter)).to eq 1
```

Comment:
Reaching into the object like this seems kinda hacky, amirite?

I would prefer exposing the `@bye_counter` with `attr_reader`. Some folks don't like to modify the implementation just to make testing easier, but I say if it makes testing easier, it makes the object easier to use in general. Future requirements of the application might benefit from it.

There's another way. You could inject the implicit dependency when `exit` is called like so:

```ruby
class Grandma
  def initialize(terminator = Kernel)
    @bye_counter = 0
    @terminator = terminator
  end
end
...
    @terminator.exit
...
Let me know if you'd like me to explain how that makes testing it easier.
```